src/app/cli/src/commands/sql_shell_command.rs

```diff
@@ -124,17 +129,30 @@ impl SqlShellCommand {
 
         Ok(())
     }
+
+    async fn run_datafusion_cli_command(&self) -> Result<(), CLIError> {
+        let mut print_options = PrintOptions {
+            format: PrintFormat::Table,
+            quiet: true,
+            maxrows: MaxRows::Limited(DEFAULT_MAX_ROWS_FOR_OUTPUT),
+        };
+
+        let mut ctx = self.query_svc.create_session().await.unwrap();
+
+        exec::exec_from_repl(&mut ctx, &mut print_options)
+            .await
+            .map_err(CLIError::failure)?;
+
+        Ok(())
+    }
 }
 
 #[async_trait::async_trait(?Send)]
 impl Command for SqlShellCommand {
     async fn run(&mut self) -> Result<(), CLIError> {
         match (self.engine.as_deref(), &self.command, &self.url) {
-            (Some("datafusion"), Some(_), None) => self.run_datafusion_command().await,
-            (Some("datafusion"), _, _) => Err(CLIError::usage_error(
-                "DataFusion engine currently doesn't have a shell and only supports single \
-                 command execution",
-            )),
+            (None, None, None) => self.run_datafusion_cli_command().await,
+            (None, Some(_), None) => self.run_datafusion_command().await,
```

Comment:
First match should be `None | Some("datafusion")` since we still want to support cases when the engine is specified explicitly, e.g.:
`kamu sql --engine datafusion`

Reply:
how about (Some("datafusion"), None, None) | (None, None, None) => self.run_datafusion_cli_command().await - ?

Reply:
an even better option might be to move what the default engine is outside of the match, e.g.:
```rust
let engine = self.engine.as_deref().unwrap_or("datafusion");
match (....) {
  ("datafusion", _, _) => ...,
  ("spark", _, _) => ...,
}
```

Reply:
You fixed it in one place but the same problem exists for `run_datafusion_command`

Reply:
On it

Reply:
done